Library/Homebrew/cli_parser.rb

```diff
@@ -0,0 +1,280 @@
+module HomebrewCli
+  class Parser
+    attr_accessor :ignore_invalid_options
+    attr_reader :unnamed
+
+    def initialize(&block)
+      @specs = {}
+      @long = {}
+      @short = {}
+      @order = []
+      @constraints = []
```

Comment:
Not obvious to me from names what any of these are unfortunately. I'd make their names much longer or: could this perhaps use an ordered array of hashes?